ratatool-sampling/src/main/scala/com/spotify/ratatool/samplers/BigSampler.scala

```diff
@@ -450,35 +640,67 @@ private[samplers] object BigSamplerBigQuery {
     } else {
       log.info(s"Will sample from BigQuery table: $inputTbl, output will be $outputTbl")
       val schema = patchedBigQueryService.getTable(inputTbl).getSchema
+      val schemaStr = JsonSerDe.toJsonString(schema)
 
-      if (fields.isEmpty) {
-        val r = sc.bigQueryTable(inputTbl)
-          .sample(withReplacement = false, samplePct)
-          .saveAsBigQuery(outputTbl, schema, WRITE_EMPTY, CREATE_IF_NEEDED, tableDescription = "")
-        sc.close().waitUntilDone()
-        r
-      } else {
-        val fieldsMissing = fields.filterNot(f => fieldInTblSchema(schema.getFields.asScala, f))
-        if (fieldsMissing.nonEmpty) {
-          throw new NoSuchElementException(
-            s"""Could not locate fields ${fieldsMissing.mkString(",")}""" +
-              s""" in table $inputTbl with schema $schema""")
-        }
-        // defeat closure TableSchema and TableFieldSchema are not serializable
-        val schemaStr = JsonSerDe.toJsonString(schema)
-        @transient lazy val schemaFields =
-          JsonSerDe.fromJsonString(schemaStr, classOf[TableSchema]).getFields.asScala
-        val samplePct100 = samplePct * 100.0
-        val r = sc.bigQueryTable(inputTbl)
-          .flatMap { e =>
+      @transient lazy val schemaFields =
+        JsonSerDe.fromJsonString(schemaStr, classOf[TableSchema]).getFields.asScala
+      @transient lazy val logSerDe = LoggerFactory.getLogger(this.getClass)
+
+      val coll = sc.bigQueryTable(inputTbl)
+
+      val sampledCollection: SCollection[TableRow] = (fields, distribution, exact) match {
+        case (Seq(), None, false) => coll.sample(withReplacement = false, fraction)
```

Comment:
I'm assuming uniform distribution is the default here (distribution = None), but maybe this could be clarified?